### PR TITLE
Feature/docker

### PR DESCRIPTION
**Pull request by phoenigm**

### CODE REVIEW

File: step3_run_apps.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
```
Comment (Dimanaux) [inline step3_run_apps.sh]: This file does nothing.

File: step2_build_dockers.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
```
Comment (Dimanaux) [inline step2_build_dockers.sh]: This file does nothing.

File: step1_build_jars.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
```
Comment (Dimanaux) [inline step1_build_jars.sh]: You just run mvn install, why do you need this file?

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+server:
```
Comment (Dimanaux) [inline src/main/resources/application.yml]: Noise changes.

File: pom.xml [hunk 5]
```diff
@@ -5,14 +5,14 @@
...
-    <artifactId>wish_list</artifactId>
+    <artifactId>wish-list</artifactId>
     <version>0.1</version>
     <packaging>jar</packaging>
 
     <parent>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-starter-parent</artifactId>
-        <version>2.0.5.RELEASE</version>
+        <version>2.2.5.RELEASE</version>
```
Comment (Dimanaux) [inline pom.xml]: No need to update spring boot.

File: pom.xml [hunk 6]
```diff
@@ -5,14 +5,14 @@
     <modelVersion>4.0.0</modelVersion>
 
     <groupId>ru.itis</groupId>
-    <artifactId>wish_list</artifactId>
+    <artifactId>wish-list</artifactId>
```
Comment (Dimanaux) [inline pom.xml]: Why renaming the project?

File: src/main/resources/application.yml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  port: 8080
+spring:
+  datasource:
+    driver-class-name: org.postgresql.Driver
+    password: postgres
+    username: postgres
+    url: jdbc:postgresql://db-wish-list/wish-list
+  jpa:
```
Comment (AinaArd) [inline src/main/resources/application.yml]: No need to mave to yml file. Properties are still good

File: pom.xml (hunk 5 above)
Comment (AinaArd) [inline pom.xml]: What's the reason?

File: README.md [hunk 8]
```diff
@@ -10,6 +10,17 @@ Wish list - web app, where users can:
 
 Project requirements:
 - Maven 3+
-- Postgres 9+
 - Java 8
+- Docker 18
```
Comment (Dimanaux) [inline README.md]: It is hard to install, why do we need it? Project doesn't have many system dependencies. So docker doesn't solve any problem. But it is much harder to maintain than PostgreSQL.

File: Dockerfile [hunk 9]
```diff
@@ -0,0 +1,4 @@
+FROM adoptopenjdk/openjdk11:alpine-jre
```
Comment (Dimanaux) [inline Dockerfile]: As you may see, the project uses Java 8.

File: .gitignore [hunk 10]
```diff
@@ -1,17 +1,161 @@
-*.class
+
```
Comment (Dimanaux) [inline .gitignore]: Changes are not related to branch target.

File: step9_stop.sh [hunk 11]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
```
Comment (Dimanaux) [inline step9_stop.sh]: You are trying to solve problems you've just created.

File: src/main/resources/application.yml (hunk 7 above)
Comment (phoenigm) [reply src/main/resources/application.yml]: Yaml is pretty and support features of collections, multi-line strings, etc

File: pom.xml (hunk 5 above)
Comment (phoenigm) [reply pom.xml]: https://github.com/spring-projects/spring-boot/issues/20217

File: step3_run_apps.sh (hunk 1 above)
Comment (phoenigm) [reply step3_run_apps.sh]: This script run apps

File: step2_build_dockers.sh (hunk 2 above)
Comment (phoenigm) [reply step2_build_dockers.sh]: This script builds dockers

File: step1_build_jars.sh (hunk 3 above)
Comment (phoenigm) [reply step1_build_jars.sh]: Same format with other scripts

File: pom.xml (hunk 5 above)
Comment (phoenigm) [reply pom.xml]: bugfix https://github.com/spring-projects/spring-boot/issues/20217

File: pom.xml (hunk 6 above)
Comment (phoenigm) [reply pom.xml]: best practices of maven module naming

File: README.md (hunk 8 above)
Comment (phoenigm) [reply README.md]: I don't want to install Postgres for running this project. I don't want to install any other application on which the project may depend.

File: Dockerfile (hunk 9 above)
Comment (phoenigm) [reply Dockerfile]: downgraded :1st_place_medal:

File: step9_stop.sh (hunk 11 above)
Comment (phoenigm) [reply step9_stop.sh]: I can run apps in docker? yes

File: step3_run_apps.sh (hunk 1 above)
Comment (Dimanaux) [reply step3_run_apps.sh]: > This script runs apps

I think `docker-compose up -d` does

File: step2_build_dockers.sh (hunk 2 above)
Comment (Dimanaux) [reply step2_build_dockers.sh]: `docker-compose build` does

File: src/main/resources/application.yml (hunk 7 above)
Comment (Dimanaux) [reply src/main/resources/application.yml]: > Yaml is pretty and support features of collections, multi-line strings, etc

Overkill

File: pom.xml (hunk 6 above)
Comment (Dimanaux) [reply pom.xml]: I don't think this is a module, any proof?

File: step9_stop.sh (hunk 11 above)
Comment (Dimanaux) [reply step9_stop.sh]: So you can run use microservices, why not to use them here?

File: README.md (hunk 8 above)
Comment (Dimanaux) [reply README.md]: So main contibutor don't want to install docker.

File: pom.xml (hunk 5 above)
Comment (Dimanaux) [reply pom.xml]: Approved.

File: step1_build_jars.sh (hunk 3 above)
Comment (Dimanaux) [reply step1_build_jars.sh]: Move then all to `bin/` directory, and name them in more adequate way
